### ISSUE TITLE
boost::transform_primary  Symbol not found after upgrading ray on MacOS

### ISSUE THREAD

**Issue by yaroslavvb** [ray-project/ray#1597]
I just upgraded my macos ray as follows

```
pip install -U https://s3-us-west-2.amazonaws.com/ray-wheels/73be23570145fbabe01ea7c83262860c21c35f06/ray-0.3.1-cp36-cp36m-macosx_10_6_intel.whl
```

Things now fail with 
```
    import pyarrow  # noqa: F401
  File "/Users/yaroslav/anaconda/envs/cifar/lib/python3.6/site-packages/ray/pyarrow_files/pyarrow/__init__.py", line 32, in <module>
    from pyarrow.lib import cpu_count, set_cpu_count
ImportError: dlopen(/Users/yaroslav/anaconda/envs/cifar/lib/python3.6/site-packages/ray/pyarrow_files/pyarrow/lib.cpython-36m-darwin.so, 2): Symbol not found: __ZNK5boost16re_detail_10650131cpp_regex_traits_implementationIcE17transform_primaryEPKcS4_
  Referenced from: /Users/yaroslav/anaconda/envs/cifar/lib/python3.6/site-packages/ray/pyarrow_files/pyarrow/libparquet.1.dylib
  Expected in: /usr/local/opt/boost/lib/libboost_regex-mt.dylib
 in /Users/yaroslav/anaconda/envs/cifar/lib/python3.6/site-packages/ray/pyarrow_files/pyarrow/libparquet.1.dylib
```

**Comment by yaroslavvb** [comment]
BTW, I downgraded on MacOS by `pip uninstall ray; pip install ray` and visualization crashes because some version_info tuple has a None 

```
import os
import ray
import ray.experimental.ui as ui

ray.init(redis_address=os.environ["REDIS_ADDRESS"])

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-664a347b3f16> in <module>()
      3 import ray.experimental.ui as ui
      4 
----> 5 ray.init(redis_address=os.environ["REDIS_ADDRESS"])

/Users/yaroslav/anaconda/lib/python3.5/site-packages/ray/worker.py in init(redis_address, node_ip_address, object_id_seed, num_workers, driver_mode, redirect_output, num_cpus, num_gpus, resources, num_custom_resource, num_redis_shards, redis_max_clients, plasma_directory, huge_pages, include_webui)
   1440                  plasma_directory=plasma_directory,
   1441                  huge_pages=huge_pages,
-> 1442                  include_webui=include_webui)
   1443 
   1444 

/Users/yaroslav/anaconda/lib/python3.5/site-packages/ray/worker.py in _init(address_info, start_ray_local, object_id_seed, num_workers, num_local_schedulers, object_store_memory, driver_mode, redirect_output, start_workers_from_local_scheduler, num_cpus, num_gpus, resources, num_redis_shards, redis_max_clients, plasma_directory, huge_pages, include_webui)
   1367             "webui_url": address_info["webui_url"]}
   1368     connect(driver_address_info, object_id_seed=object_id_seed,
-> 1369             mode=driver_mode, worker=global_worker, actor_id=NIL_ACTOR_ID)
   1370     return address_info
   1371 

/Users/yaroslav/anaconda/lib/python3.5/site-packages/ray/worker.py in connect(info, object_id_seed, mode, worker, actor_id)
   1762     except Exception as e:
   1763         if mode in [SCRIPT_MODE, SILENT_MODE]:
-> 1764             raise e
   1765         elif mode == WORKER_MODE:
   1766             traceback_str = traceback.format_exc()

/Users/yaroslav/anaconda/lib/python3.5/site-packages/ray/worker.py in connect(info, object_id_seed, mode, worker, actor_id)
   1759     # information that the Ray cluster was started with.
   1760     try:
-> 1761         ray.services.check_version_info(worker.redis_client)
   1762     except Exception as e:
   1763         if mode in [SCRIPT_MODE, SILENT_MODE]:

/Users/yaroslav/anaconda/lib/python3.5/site-packages/ray/services.py in check_version_info(redis_client)
    357                         "    Python: " + version_info[2] + "\n"
    358                         "    Cloudpickle: " + version_info[3] + "\n"
--> 359                         "    Pyarrow: " + version_info[4])
    360 
    361 

TypeError: Can't convert 'NoneType' object to str implicitly
```

**Comment by robertnishihara** [comment]
Thanks for the bug report.

So the original problem was solved by downgrading to `0.3.0`? Is `0.3.1` still not working for you? The version string problem should be fixed in the current master.

**Comment by yaroslavvb** [comment]
I can try latest master...what's the process again to get wheel?
I tried plugging in latest commit into url, but it's HTTP 404 not found

commit=ba1ce85f58faa4e4265dc4d102b7770236fb163a
pip install -U https://s3-us-west-2.amazonaws.com/ray-wheels/$commit/ray-0.3.1-cp36-cp36m-macosx_10_6_intel.whl

**Comment by robertnishihara** [comment]
Ah well our wheels haven't been building for a handful of commits due to #1587. Should be working again by tomorrow.

**Comment by robertnishihara** [comment]
Ok, I'm seeing the original error as well now on Mac.

**Comment by robertnishihara** [comment]
Should be fixed by #1688.
